rs/nns/governance/canister/governance.did

```diff
@@ -491,10 +505,21 @@ type ManageNeuronRequest = record {
   id : opt NeuronId;
   command : opt ManageNeuronCommandRequest;
   neuron_id_or_subaccount : opt NeuronIdOrSubaccount;
+
+  // Controls how the resulting_neuron field in the response is populated.
+  // Possible values here and how the response will be populated:
+  //
+  //     * null - Not populated.
+  //     * 1 - NeuronInfo
+  //     * 2 - NueronInfo, but "big" fields are cleared. To wit,
+  //         * recent_ballots
+  //         * known_neuron_data
+  resulting_neuron_info : opt int32;
 };
 
 type ManageNeuronResponse = record {
   command : opt Command_1;
+  resulting_neuron_info : opt NeuronInfo;
```

Comment:
Does this mean that for each manage neuron call the neuronInfo can newly be included? 
How do we decide in which answers it will be included? Just for the commands where someone asked for it?

Reply:
This is a list of requests we have so far:
* indication of VP to response for claim_or_refresh
* indication of VP to response for cast_vote (in the case have 0 VP)

Reply:
> Does this mean that for each manage neuron call the neuronInfo can newly be included?

Yes.

> How do we decide in which answers it will be included?

This is populated when the request has resulting_neuron_info = 1 or 2.

> Just for the commands where someone asked for it?

Yes.

Reply:
> when the request has resulting_neuron_info = 1 or 2.

TBH, this is anti-[KISS]. A simpler way would be to just always embed NeuronInfo without clearing anything (behavior 1). The problem with this approach is that there are some large fields in NeuronInfo, and they are often not needed, wasting instructions, network, space, time, etc.

[KISS]: https://en.wikipedia.org/wiki/KISS_principle

This also probably violates

> Premature optimization is the root of all evil.
> --Donald Knuth

What we can instead do is get rid of the request field that controls the behavior of this field, and if we later find there is a need to optimize, we can introduce the request field. The problem is that if there is a need, simply adding the request field later is not enough to save resources. We would actually need clients to use the new field to request less data. Otherwise, adding the field would not be backwards compatible.

Reply:
Lara and I resolved over zoom that quill can probably solve the problem of warning the user when their neurons need to be refreshed soon (or now) without this. Will ask Adam to confirm. For now, we optimistically go with the assumption that no backend changes are needed to solve that problem.